sdk/iot/hub/tests/test_az_iot_hub_client.c

```diff
@@ -880,6 +934,9 @@ int test_iot_hub_client()
     cmocka_unit_test(test_az_iot_hub_client_get_user_name_small_buffer_fail),
     cmocka_unit_test(test_az_iot_hub_client_get_user_name_user_options_succeed),
     cmocka_unit_test(test_az_iot_hub_client_get_user_name_user_options_small_buffer_fail),
+    cmocka_unit_test(test_az_iot_hub_client_get_user_name_with_model_id_succeed),
```

Comment:
could you add pre-condition checks and a withUsegAgentTest please?  Even though we think this is short-term code, it may end up in permanent SDK so would like a bit of extra guards.

Reply:
yea will do

Reply:
done. four precondition and an additonal "max options" case